lib/relation-definition.js

```diff
@@ -1714,6 +1785,19 @@ HasOne.prototype.related = function (refresh, params) {
   }
 };
 
+/**
+ * Define a Promise-based method for the hasOne relation itself
+ * - order.customer.get(refresh): Load the target model instance asynchronously
+ *
+ * @param {Boolean} refresh Reload from the data source
+ * @returns {Promise}
+ */
+
+HasOne.prototype.get = function (refresh) {
+  var cb = utils.createPromiseCallback();
+  return this.related(refresh, cb);
```

Comment:
Seems to uncover a bug with hasOne... it's failing with promises, and if I change the normal test to use `refresh=true`, it also fails:
```
  1) relations hasOne with scope can be used to query data:
     Uncaught TypeError: Cannot read property 'id' of null
```